httpx/http_service.go

```diff
@@ -126,7 +149,8 @@ func (t *RetryTransport) RoundTrip(ctx context.Context, req *http.Request) (*htt
 		if resp.StatusCode >= 500 {
 			return nil, &RetryableHTTPError{Path: req.URL.String(), StatusCode: resp.StatusCode}
```

Comment:
Since the return value `return nil, &RetryableHTTPError{Path: req.URL.String(), StatusCode: resp.StatusCode}` could end up being returned to the caller in some cases, should this line return a non-nil response as well?